lib/project/potion_dialog/potion_dialog.rb

```diff
@@ -0,0 +1,42 @@
+class PotionDialog
+
+  def initialize(options)
+    # err if missing required options
+    raise "Requires an xml_layout!!" unless options[:xml_layout]
+    raise "Cannot have width without height!!" if options[:width] && !options[:height]
+    raise "Cannot have height without width!!" if options[:height] && !options[:width]
+    
```

Comment:
All exceptions starts with [BluePotion ERROR]. Also in the message note that it is PotionDialog#initialize. RM Android doesn't report well where things happen. Leave off the !! please.